public/scripts/main.js

```diff
@@ -0,0 +1,65 @@
+import * as ai from "./ai";
+import * as request from "./request";
+import * as statusDisplay from "./statusDisplay";
+
+const buttons = document.querySelectorAll("button");
+const markedClassName = "marked";
+const spaces = document.querySelectorAll(".space");
+
+let state = {}
```

Comment:
Is there a way to refactor this so that you don't need to maintain global state?

Reply:
Because I need to pass the entire game state to the backend, I'm not really sure how to avoid maintaining global state. I could probably make use of JavaScript classes to store that information, but I would still need to maintain a global reference pointing to an instance of that class. That doesn't seem like a much better option to me. Do you have any suggestions?

Reply:
Because I need to pass the entire game state to the backend, I'm not sure how I'd avoid maintaining global state. I could probably make use of JavaScript classes, but I think I'd still need to maintain a global reference to an instance of that class. I'm not sure that's a much better solution. Do you have any suggestions?